contracts/src/receipt.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright 2025 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use alloy_primitives::Bytes;
+use risc0_aggregation::{
+    decode_set_inclusion_seal, SetInclusionDecodingError, SetInclusionEncodingError,
+    SetInclusionReceipt,
+};
+use risc0_zkvm::{FakeReceipt, InnerReceipt, Receipt as Risc0Receipt, ReceiptClaim};
+use thiserror::Error;
+
+use crate::{
+    encode_seal,
+    groth16::decode_groth16_seal,
+    selector::{SelecorType, Selector, SelectorError},
+};
+
+/// Wrapper around different types of receipts.
+pub enum ReceiptType {
+    Risc0(Risc0Receipt),
+    SetInclusion(SetInclusionReceipt<ReceiptClaim>),
+}
+
+impl ReceiptType {
+    /// Encode the receipt as a seal.
+    pub fn encode_seal(&self) -> Result<Vec<u8>, SetInclusionEncodingError> {
```

Comment:
To be a bit more specific. Folks might be looking for ABI as a keyword as well.
```suggestion
    pub fn abi_encode_seal(&self) -> Result<Vec<u8>, SetInclusionEncodingError> {
```